source/deprecations/v2.x.html.md

```diff
@@ -753,3 +753,92 @@ component.appendTo('.my-component-wrapper');
 ```
 
 Note that both APIs are private, so no public API is been deprecated here.
+
+
+#### `{{render` helper
+
+##### until: 3.0.0
+##### id: ember-template-compiler.deprecate-render
+
+Using the `{{render` helper is deprecated in favor of using components.
+Please refactor to a component and invoke thusly:
+
+For example, if you had:
+
+```hbs
+{{render 'my-sidebar'}}
+```
+
+```hbs
+{{! app/templates/my-sidebar.hbs }}
+<p>template stuff here</p>
+```
+
+```js
+// app/controllers/my-sidebar.js
+export default Ember.Controller.extend({
+});
+```
+
+Would be refactored to:
+
+```hbs
+{{my-sidebar}}
+```
+
+```hbs
+{{! app/templates/components/my-sidebar.hbs }}
+<p>template stuff here</p>
+```
+
+```js
+// app/components/my-sidebar.js
+export default Ember.Component.extend({
+});
+```
+
+#### Rendering into a {{render}} helper that resolves to an {{outlet}}.
+
+##### until: 3.0.0
+##### id: ember-routing.top-level-render-helper
+
+For example, if you had:
+
+```hbs
+{{! app/templates/application.hbs }}
+{{render "sidebar"}}
+```
+
+```hbs
+{{! app/templates/sidebar.hbs }}
+<div class="sidebar">{{outlet}}</div>'
+```
+
+```hbs
+{{! app/templates/index.hbs }}
+Index page
+```
+
+```js
+// app/controllers/my-sidebar.js
+App.IndexRoute = Ember.Route.extend({
+  renderTemplate() {
+    this.render({ into: 'sidebar' });
+  },
+  actions: {
+    disconnect: function() {
```

Comment:
shorthand function syntax was used for `renderTemplate`, the same should be used here. `disconnect() {`